subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/GradleVsMavenPerformanceTestRunner.groovy

```diff
@@ -89,34 +78,40 @@ class GradleVsMavenPerformanceTestRunner extends CrossBuildGradleProfilerPerform
 
     protected void finalizeSpec(BuildExperimentSpec.Builder builder) {
         super.finalizeSpec(builder)
-        if (builder instanceof GradleBuildExperimentSpec.GradleBuilder) {
-            def invocation = (GradleInvocationSpec.InvocationBuilder) builder.invocation
-            invocation.gradleOptions = customizeJvmOptions(invocation.gradleOptions)
-            if (!builder.displayName.startsWith("Gradle ")) {
-                throw new IllegalArgumentException("Gradle invocation display name must start with 'Gradle '")
-            }
-        } else if (builder instanceof MavenBuildExperimentSpec.MavenBuilder) {
-            def invocation = ((MavenBuildExperimentSpec.MavenBuilder) builder).invocation
-            invocation.jvmOpts = customizeJvmOptions(invocation.jvmOpts)
-            if (!invocation.args.find { it.startsWith("-Dmaven.repo.local=") }) {
-                def localRepoPath = m2.mavenRepo().rootDir.absolutePath
-                if (OperatingSystem.current().isWindows()) {
-                    localRepoPath = localRepoPath.replace("\\", "\\\\").replace(" ", "\\ ")
-                    invocation.args.add("-Dmaven.repo.local=${localRepoPath}".toString())
-                } else {
-                    invocation.args.add("-Dmaven.repo.local=${localRepoPath}".toString())
-                }
-            }
-            if (!invocation.mavenHome) {
-                def home = System.getProperty("MAVEN_HOME")
-                if (home) {
-                    invocation.mavenHome(new File(home))
-                }
+        if (builder instanceof MavenBuildExperimentSpec.MavenBuilder) {
+            finalizeMavenBuildSpec(builder)
+        }
+    }
+
+    private void finalizeMavenBuildSpec(MavenBuildExperimentSpec.MavenBuilder builder) {
+        def invocation = builder.invocation
+        invocation.jvmOpts = customizeJvmOptions(invocation.jvmOpts)
+        if (!invocation.args.find { it.startsWith("-Dmaven.repo.local=") }) {
+            def localRepoPath = m2.mavenRepo().rootDir.absolutePath
+            if (OperatingSystem.current().isWindows()) {
+                localRepoPath = localRepoPath.replace("\\", "\\\\").replace(" ", "\\ ")
+                invocation.args.add("-Dmaven.repo.local=${localRepoPath}".toString())
+            } else {
+                invocation.args.add("-Dmaven.repo.local=${localRepoPath}".toString())
             }
-            if (!builder.displayName.startsWith("Maven ")) {
-                throw new IllegalArgumentException("Maven invocation display name must start with 'Maven '")
+        }
+        if (!invocation.mavenHome) {
+            def home = System.getProperty("MAVEN_HOME")
```

Comment:
I changed it to `System.getenv()`.

Reply:
I removed this completely. It seems like we didn't pass the Maven installation to the actual Maven invocation, since Gradle profiler doesn't have an API for it. I fixed it by setting `MAVEN_HOME`: https://github.com/gradle/gradle/pull/14441/commits/6c09357fbda103067a696fba25ce613bf975d69f